Mathlib/Analysis/Convex/Between.lean

```diff
@@ -70,8 +70,11 @@ theorem right_mem_affineSegment (x y : P) : y ∈ affineSegment R x y :=
 theorem affineSegment_same (x : P) : affineSegment R x x = {x} := by
   -- porting note: added as this doesn't do anything in `simp_rw` any more
   rw [affineSegment]
+  -- Note: when adding "simp made no progress" in lean4#2336,
+  -- somehow this `simp` doesn't work in `simp_rw`.
+  simp only [lineMap_same]
   -- porting note: added `_ _` and `Function.const`
-  simp_rw [lineMap_same, AffineMap.coe_const _ _, Function.const,
+  simp_rw [AffineMap.coe_const _ _, Function.const,
```

Comment:
I'll copy the comment I just made at #6019 to here:

`simp_rw [lineMap_same _]` seems to work, but I don't quite understand why.